eng/common/scripts/Package-Properties.ps1

```diff
@@ -130,6 +130,17 @@ function Get-PrPkgProperties([string]$InputDiffJson) {
         {
             $filePath = Resolve-Path (Join-Path $RepoRoot $file)
             $shouldInclude = $filePath -like "$pkgDirectory*"
+            $pathComponents = $file -split "/"
+
+            # sdk/<service>/<file.extension>
+            if ($pathComponents.Length -eq 3 -and $pathComponents[0] -eq "sdk") {
+                $servicePkgs = $allPackageProperties | Where-Object { $_.ServiceDirectory -eq $pathComponents[1] } | ForEach-Object { $_.Name }
```

Comment:
Do we really need to look at all packages here given we are in the context of the outer loop should we just check if that package is in the same service directory?

Reply:
I squinted at this for a couple minutes. I originally wrote this when it was a in a second pass over just the changed service files. You're absolutely right that we're going to add multiple duplicates by invoking this.

Honestly though @weshaggard , I'm leaning towards moving this to the `Language-Settings` specific, where specific languages will be able to customize this behavior.